adafruit_circuitplayground/circuit_playground_base.py

```diff
@@ -566,6 +565,9 @@ def button_a(self):
               if cp.button_a:
                   print("Button A pressed!")
         """
+        if self._a is None:
+            self._a = digitalio.DigitalInOut(board.BUTTON_A)
+            self._a.switch_to_input(pull=digitalio.Pull.DOWN)
```

Comment:
You _could_ refactor this into:
````
    @staticmethod
    def _button_setup(pin):
        b = digitalio.DigitalInOut(pin)
        b.switch_to_input(pull=digitalio.Pull.DOWN)
        return b
...
    if self._a is None:
        self._a = self._button_setup(board.BUTTON_A)
```
and same for B. The same routine (maybe change the name) could be used for the `self._switch` setup above (lines 60-61) as well.

Reply:
The switch requires a pullup, not a pulldown like the buttons. So it couldn't be reused exactly as is for the switch. 

Do you think this is better Python to refactor it? Because in terms of what it's adding or subtracting, it seems to be a wash on it from the current way it's done, e.g. refactoring it this way appears to remove 4 lines of code to add 4 lines of code. I'm happy to refactor it if you think it's better that way, but I want to understand the reasoning.

Reply:
I missed the switch difference, sorry. It doesn't really matter. It might be a tiny bit smaller when compiled. I'm happy to approve as is.